### PR TITLE
chore(docker): Fixes supporting more deployment flexibility

### PR DESCRIPTION
**Pull request by scottwittenburg**
Allow presence of env vars "SERVER_NAME" and "PROTOCOL" to take
precedence when defining the session url root to rewrite into the
launcher config.

Also allow the "addEndpoints" script to look for endpoint name with
special form "DOCUMENT-ROOT-DIRECTORY" to use in rewriting the main
DocumentRoot path in the apache configuration.

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@jourdain Want to take a look?  I added a minimum bit of documentation in the `tools/docker/paraviewweb/Dockerfile` regarding the "fully external" stuff.  We could update the online docs later.

File: tools/docker/paraviewweb/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,53 @@
...
+#
+# In order to run this image directly as a paraviewweb application, there are
+# a couple assumptions we make about the external directories.  You should first
+# create a directory structure where your application lives (we'll call it
+# <APP_ROOT> below) following this structure:
+#
+#     <APP_ROOT>/
+#     <APP_ROOT>/www/
+#     <APP_ROOT>/launcher/config.json
+#
```
Comment (jourdain) [inline tools/docker/paraviewweb/Dockerfile]: We are missing the must-be values. (i.e.: proxy-map file path, possible log files inside docker, pvpython path inside the docker for the cmd part...)

File: documentation/content/docs/docker.md [hunk 2]
```diff
@@ -184,3 +184,38 @@ sudo docker run --runtime=nvidia             \
...
+
+```
+export SERVER_NAME="localhost:9000"
+export PROTOCOL="ws"
+export EXTRA_PVPYTHON_ARGS="-dr,--mesa-swr"
+
+sudo docker run --runtime=nvidia             \
+    -p 0.0.0.0:9000:80                        \
+    -v ~:/data                                 \
+    -ti kitware/paraviewweb:pvw-egl-demo-v5.6.0
```
Comment (jourdain) [inline documentation/content/docs/docker.md]: Should this be done with the `-e` argument so those env variables are indeed passed to docker?

```
sudo docker run --runtime=nvidia             \
    -p 0.0.0.0:9000:80                        \
    -v ~:/data                                 \
    -e SERVER_NAME="localhost:9000" \
    -e PROTOCOL="ws" \
    -e EXTRA_PVPYTHON_ARGS="-dr,--mesa-swr" \
    -ti kitware/paraviewweb:pvw-egl-demo-v5.6.0
```

Comment (scottwittenburg) [reply documentation/content/docs/docker.md]: Yep, oversight on my part.

**Comment by scottwittenburg** [comment]
@jourdain I'm just about ready to rebuild the pvw base, demo, and visualizer images using these changes.  But I can wait until you have a chance to take another look at this first.

**Comment by jourdain** [comment]
Looks good. We might be just missing a line or two on the fact you can install other python modules using a `requirements.txt` file when placed inside `<APP_ROOT>`

**Pull request merged**